### PR TITLE
started Auth

### PR DESCRIPTION
**Pull request by yogesh-**
Completed Authentication with following features -
- Login
- Guest Login
- Signup as New user
- Log Out user

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
code looks good to be merged

File: src/components/navbar/navbar.jsx [hunk 1]
```diff
@@ -3,10 +3,29 @@ import { Link } from "react-router-dom";
...
+import { useAuth } from "../../context/authContext";
 
 export const Navbar = () => {
+  const { setIsLoggedIn } = useAuth();
   const { cartState } = useCart();
   const { wishState } = useWish();
+  let navigate = useNavigate();
+
+  let navtoken = localStorage.getItem("token");
+  // console.log("navtoken", navtoken);
```
Comment (tushargoyal812) [inline src/components/navbar/navbar.jsx:16]: remove comments

File: src/components/navbar/navbar.jsx [hunk 2]
```diff
@@ -3,10 +3,29 @@ import { Link } from "react-router-dom";
...
+
+  const logoutLoginHandler = () => {
+    if (navtoken) {
+      // delete-token,log out the user,go to home
+      setIsLoggedIn(false);
+      localStorage.removeItem("token");
+      navigate("/");
+      // setIsLoggedIn(false);
```
Comment (tushargoyal812) [inline src/components/navbar/navbar.jsx:24]: remove comments

File: src/pages/login/login.jsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+export const Login = () => {
+  let navigate = useNavigate();
+
+  const [isToken, setIsToken] = useState(false);
+  const [loginEmail, setLoginEmail] = useState();
+  const [loginPwd, setLoginPwd] = useState();
+
+  let userEmail = localStorage.getItem("userEmail");
+  let userPwd = localStorage.getItem("userPwd");
+  console.log("getUser", userEmail, userPwd);
```
Comment (tushargoyal812) [inline src/pages/login/login.jsx:18]: remove console

File: src/pages/login/login.jsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const loginHandlerGuest = async () => {
+    try {
+      let data = {
+        email: "adarshbalika@gmail.com",
+        password: "adarshBalika123",
+      };
+      let loginapi = await axios.post("/api/auth/login", data);
+
+      console.log(loginapi.data.encodedToken);
```
Comment (tushargoyal812) [inline src/pages/login/login.jsx:47]: remove console

File: src/pages/signup/signup.jsx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            email: inputData.email,
+            password: inputData.password,
+          });
+
+          localStorage.setItem("userEmail", response.data.createdUser.email);
+          localStorage.setItem("userPwd", response.data.createdUser.password);
+          // let userEmail = localStorage.getItem("userEmail");
+          // let userPwd = localStorage.getItem("userPwd");
+          // console.log("getUser", userEmail, userPwd);
+          // saving the encodedToken in the localStorage
```
Comment (tushargoyal812) [inline src/pages/signup/signup.jsx:74]: remove comments

File: src/pages/signup/signup.jsx [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+          localStorage.setItem("token", response.data.encodedToken);
+          let signToken = localStorage.getItem("token");
+          if (signToken) {
+            navigate("/");
+          }
+          console.log(signToken);
```
Comment (tushargoyal812) [inline src/pages/signup/signup.jsx:80]: remove console

File: src/components/navbar/navbar.jsx (hunk 1 above)
Comment (yogesh-) [reply src/components/navbar/navbar.jsx:16]: done

File: src/pages/login/login.jsx (hunk 3 above)
Comment (yogesh-) [reply src/pages/login/login.jsx:18]: done

**Pull request merged**